### PR TITLE
use port-mapping endpoint for kind clusters on Docker Desktop

### PR DESCRIPTION
**Pull request by bartdong**
Signed-off-by: bartdong [bartdong@tencent.com](mailto:bartdong@tencent.com)

#### What type of PR is this?
kind/bug

#### What this PR does / why we need it:
When running with `Docker Desktop`, the parent cluster apiserver is not accessible from Mac host.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: hack/local-running.sh [hunk 1]
```diff
@@ -36,7 +36,7 @@ function create_cluster() {
   kind create cluster --name "${cluster_name}" --kubeconfig="${kubeconfig}" --image="${image}" 2>&1
 
   kubectl config rename-context "kind-${cluster_name}" "${cluster_name}" --kubeconfig="${kubeconfig}"
-  kind_server="https://$(docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' "${cluster_name}-control-plane"):6443"
+  kind_server="https://$(docker inspect --format='{{(index (index .NetworkSettings.Ports "6443/tcp") 0).HostIp}}:{{(index (index .NetworkSettings.Ports "6443/tcp") 0).HostPort}}' "${cluster_name}-control-plane")"
```
Comment (dixudx) [inline hack/local-running.sh:39]: IIRC, current script works well. Have you got any exceptions/tracing logs?

Comment (bartdong) [reply hack/local-running.sh:39]: Hi, I've been stuck at helm installing.
```
+ echo 'Installing clusternet-hub...'
Installing clusternet-hub...
+ helm --kubeconfig=/Users/bart/.kube/clusternet.config --kube-context=parent install clusternet-hub -n clusternet-system --create-namespace clusternet/clusternet-hub
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/bart/.kube/clusternet.config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/bart/.kube/clusternet.config
Error: INSTALLATION FAILED: Kubernetes cluster unreachable: Get "https://172.18.0.5:6443/version": read tcp 192.168.255.10:62049->172.18.0.5:6443: read: connection reset by peer
```

**Comment by Garrybest** [comment]
Hi @bartdong, I check this command and it returns a host IP and port.

```bash
# docker inspect --format='{{(index (index .NetworkSettings.Ports "6443/tcp") 0).HostIp}}:{{(index (index .NetworkSettings.Ports "6443/tcp") 0).HostPort}}' "parent-control-plane"
127.0.0.1:35605
```

This IP can be accessed by host, but cannot be accessed by any other `kind` clusters becasue `127.0.0.1` represents the kind container IP itself. We should get the bridge IP of kind clusters.

**Comment by dixudx** [comment]
> Hi @bartdong, I check this command and it returns a host IP and port.
> 
> ```shell
> # docker inspect --format='{{(index (index .NetworkSettings.Ports "6443/tcp") 0).HostIp}}:{{(index (index .NetworkSettings.Ports "6443/tcp") 0).HostPort}}' "parent-control-plane"
> 127.0.0.1:35605
> ```
> 
> This IP can be accessed by host, but cannot be accessed by any other `kind` clusters becasue `127.0.0.1` represents the kind container IP itself. We should get the bridge IP of kind clusters.

Exactly. This issue can be mitigated by updating `parent-contol-plane` endpoint with that port-mapping address. While the parent apiserver url used for cluster registration stays the same.

**Review by dixudx — APPROVED** [review]
looks good to me

**Review by dixudx — APPROVED** [review]

**Pull request merged**